spec/lib/publishing_api_payload/history_spec.rb

```diff
@@ -0,0 +1,110 @@
+RSpec.describe PublishingApiPayload::History do
+  describe "#public_updated_at" do
+    it "returns current time if a major change is published" do
+      freeze_time do
+        first_edition = build(:edition)
+        second_edition = build(:edition, document: first_edition.document, update_type: "major")
+
+        expect(described_class.new(second_edition).public_updated_at).to eq(Time.current)
+      end
+    end
+
+    it "returns most recent change history time if a minor change is published" do
+      first_edition = build(:edition, first_published_at: "2020-02-22 11:00:00")
+      second_edition = build(:edition, document: first_edition.document, update_type: "minor")
+
+      expect(described_class.new(second_edition).public_updated_at).to eq("2020-02-22 11:00:00")
+    end
+  end
+
+  describe "#first_published_at" do
+    it "returns backdated date if edition is backdated" do
+      edition = build(:edition, first_published_at: "2020-02-22 11:00:00", backdated_to: "2020-02-20 09:00:00")
+
+      expect(described_class.new(edition).first_published_at).to eq("2020-02-20 09:00:00")
+    end
+
+    it "returns first published date if edition is not backdated" do
+      edition = build(:edition, first_published_at: "2020-02-22 11:00:00")
+
+      expect(described_class.new(edition).first_published_at).to eq("2020-02-22 11:00:00")
+    end
+  end
+
+  describe "#change_history" do
+    let(:edition) { build(:edition, first_published_at: "2020-02-22 11:00:00") }
+
+    it "appends first change note to end of history array" do
+      expected_change_note = {
+        note: "First published.",
+        public_timestamp: "2020-02-22 11:00:00",
+      }
+
+      expect(described_class.new(edition).change_history.last).to eq(expected_change_note)
+    end
+
+    it "adds change note to start of array for major changes" do
+      freeze_time do
+        second_edition = build(:edition, document: edition.document, number: 2, update_type: "major", change_note: "Some changes")
+
+        expected_change_note = {
+          note: "Some changes",
+          public_timestamp: Time.current,
+        }
+
+        expect(described_class.new(second_edition).change_history.first).to eq(expected_change_note)
+      end
+    end
+
+    it "does not add change note to start of array for minor changes" do
+      second_edition = build(:edition, document: edition.document, number: 2, update_type: "minor")
+
+      expected_change_note = {
+        note: "First published.",
+        public_timestamp: "2020-02-22 11:00:00",
+      }
+
+      expect(described_class.new(second_edition).change_history.first).to eq(expected_change_note)
```

Comment:
I wasn't sure if this test quite captured the scenario as it only tests one of the change notes.

Might be better if the assertion was more along the lines of:

`expect(described_class.new(second_edition).change_history).to eq([expected_change_note])`

Reply:
Made that change, and also made a similar change for other tests where we expect an array to be returned by `change_history`.